compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/interop/JMethodResolver.java

```diff
@@ -897,6 +898,10 @@ private Executable resolveMethod(Class<?> clazz, String name, Class<?>... paramT
     private List<Executable> getExecutables(Class<?> clazz, String methodName, JMethodKind kind) {
 
         if (kind == JMethodKind.CONSTRUCTOR) {
+            if (Modifier.isAbstract(clazz.getModifiers())) {
```

Comment:
Should we check interfaces?

Reply:
An interface cannot contain a constructor. It will give `CONSTRUCTOR_NOT_FOUND` error when an interface is being tried to instantiate via interop calls.

Reply:
IMO for similar scenarios with interfaces, it's better to provide `INSTANTIATION_ERROR`. Since `Modifier.isAbstract(clazz.getModifiers())` returns `true` for interfaces as well, we can keep this check to cover both abstract class and interface cases.

Reply:
Lets add test case if it is not there

Reply:
Added a test case with interface.